flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTask.java

```diff
@@ -58,6 +61,7 @@
 
 	public SourceStreamTask(Environment env) {
 		super(env);
+		this.sourceThread = new LegacySourceFunctionThread(getName());
```

Comment:
Here we are calling a non-final method from the constructor :(

Reply:
The name could be set in `performDefaultAction()`, before `sourceThread.start();`

Reply:
Not the best solution, but let's not overthink this. +1 for setting the name in `performDefaultAction`